src/Forms/FieldsProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Chrometoaster\AdvancedTaxonomies\Forms;
+
+use Chrometoaster\AdvancedTaxonomies\Models\TaxonomyTerm;
+use SilverStripe\Forms\GridField\GridFieldAddExistingAutocompleter;
+use SilverStripe\Forms\GridField\GridFieldAddNewButton;
+use SilverStripe\Forms\GridField\GridFieldConfig;
+use SilverStripe\Forms\GridField\GridFieldConfig_RelationEditor;
+use SilverStripe\Forms\GridField\GridFieldDataColumns;
+use SilverStripe\Forms\GridField\GridFieldEditButton;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\ORM\DataList;
+use SilverStripe\Versioned\GridFieldArchiveAction;
+
+/**
+ * Class FieldsProvider
+ *
+ * Providing reusable fields and field configurations.
+ */
+class FieldsProvider
+{
+    /**
+     * Provide a reusable gridfield config for tagging a dataobject with some terms
+     *
+     * @param DataList|null $searchList
+     * @param array $extraDisplayFields
+     * @param string $sortField
+     * @return GridFieldConfig
+     */
+    public static function getTaggingGridFieldConfig(DataList $searchList = null, array $extraDisplayFields = [], string $sortField = 'Sort'): GridFieldConfig
+    {
+        // Remove config components from the Tags gridfield to disallow adding/deleting/archiving taxonomy terms from here
+        $gfc = GridFieldConfig_RelationEditor::create();
+        $gfc->removeComponentsByType([
+            GridFieldAddNewButton::class,
+            GridFieldEditButton::class,
+            GridFieldArchiveAction::class,
+            GridFieldFilterHeader::class,
+        ]);
+
+        $gfc->getComponentByType(GridFieldDataColumns::class)->setDisplayFields(array_merge(
+            [
+                'getNameAsTagWithExtraInfo' => 'Name',
+                'getDescription15Words'     => 'Description',
+            ],
+            $extraDisplayFields
+        ));
+
+        $gfc->addComponent(GridFieldOrderableRows::create($sortField));
```

Comment:
same for this `GridFieldOrderableRows`, we used our own custom `GridFieldOrderableRows` for that famous bug (you've submit some test case for our PR to core). Are we no longer using our custom class?